exchanges/poloniex/poloniex_wrapper.go

```diff
@@ -1030,48 +1031,46 @@ func (e *Exchange) GetOrderInfo(ctx context.Context, orderID string, pair curren
 				Timestamp: td.CreateTime.Time(),
 				Amount:    td.Quantity.Float64(),
 				Fee:       td.FeeAmount.Float64(),
-				Side:      stringToOrderSide(td.Side),
+				Side:      td.Side,
 				Type:      oType,
 			}
 		}
 		resp, err := e.GetOrder(ctx, orderID, "")
 		if err != nil {
 			smartOrders, err := e.GetSmartOrderDetails(ctx, orderID, "")
-			if err != nil {
+			switch {
+			case err != nil:
 				return nil, err
-			} else if len(smartOrders) == 0 {
+			case len(smartOrders) == 0:
 				return nil, order.ErrOrderNotFound
+			case len(smartOrders) != 1:
+				return nil, fmt.Errorf("%w: too may smart orders returned", common.ErrInvalidResponse)
 			}
-			if len(smartOrders) > 0 {
-				cp, err := currency.NewPairFromString(smartOrders[0].Symbol)
-				if err != nil {
-					return nil, err
-				}
-				if !pair.IsEmpty() && !cp.Equal(pair) {
-					return nil, fmt.Errorf("order with ID %s expected a symbol %v, but got %v", orderID, pair, smartOrders[0].Symbol)
-				}
-				oType, err := order.StringToOrderType(smartOrders[0].Type)
-				if err != nil {
-					return nil, err
-				}
-				return &order.Detail{
-					Side:          stringToOrderSide(smartOrders[0].Side),
-					Pair:          cp,
-					Exchange:      e.Name,
-					Trades:        orderTrades,
-					OrderID:       smartOrders[0].ID,
-					TimeInForce:   smartOrders[0].TimeInForce,
-					ClientOrderID: smartOrders[0].ClientOrderID,
-					Price:         smartOrders[0].Price.Float64(),
-					QuoteAmount:   smartOrders[0].Amount.Float64(),
-					Date:          smartOrders[0].CreateTime.Time(),
-					LastUpdated:   smartOrders[0].UpdateTime.Time(),
-					Amount:        smartOrders[0].Quantity.Float64(),
-					Type:          oType,
-					Status:        orderStateFromString(smartOrders[0].State),
-					AssetType:     stringToAccountType(smartOrders[0].AccountType),
-				}, nil
+			if !pair.IsEmpty() && !smartOrders[0].Symbol.Equal(pair) {
+				return nil, fmt.Errorf("order with ID %s expected a symbol %v, but got %v", orderID, pair, smartOrders[0].Symbol)
```

Comment:
🚧 Change request:
My patch removed this. You've given it a thumbs up, but left it without explanation?
This **remains inconsistent with non-smart orders below**.

Either remove, or implement for both.

I still vote, just remove it.